mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -251,106 +258,230 @@ static void populateDefaults() {
+                               cmpResultAllocOp, ValueRange{c0IndexOp});
+
+  verifyResultsBlock->push_back(loop0);
+
+  // Emit print function call
+  auto unrankedMemRefType =
+      UnrankedMemRefType::get(builder.getIntegerType(32), 0);
+  auto printMemRefCastOp = builder.create<MemRefCastOp>(
+      builder.getUnknownLoc(), cmpResultAllocOp, unrankedMemRefType);
+  auto printMemRefFuncOp =
+      FuncOp::create(builder.getUnknownLoc(), "print_memref_i32",
+                     builder.getFunctionType({unrankedMemRefType}, {}));
+  auto printMemRefCallOp =
+      builder.create<CallOp>(builder.getUnknownLoc(), printMemRefFuncOp,
+                             ValueRange{printMemRefCastOp});
+  module.push_back(printMemRefFuncOp);
+  verifyResultsBlock->push_back(printMemRefCastOp);
+  verifyResultsBlock->push_back(printMemRefCallOp);
 
-  // Emit memref_cast.
-  auto filterMemRefCastOp = builder.create<MemRefCastOp>(
-      builder.getUnknownLoc(), filterHostAllocOp, fourDimUnknownSizeMemRefType);
-  auto inputMemRefCastOp = builder.create<MemRefCastOp>(
-      builder.getUnknownLoc(), inputHostAllocOp, fourDimUnknownSizeMemRefType);
-  auto outputMemRefCastOp = builder.create<MemRefCastOp>(
-      builder.getUnknownLoc(), outputHostAllocOp, fourDimUnknownSizeMemRefType);
-  block->push_back(filterMemRefCastOp);
-  block->push_back(inputMemRefCastOp);
-  block->push_back(outputMemRefCastOp);
+  auto returnOp =
+      builder.create<ReturnOp>(builder.getUnknownLoc(), ValueRange{});
+  verifyResultsBlock->push_back(returnOp);
 
-  // Populate initial values.
-  auto oneConstantFloatOp = builder.create<ConstantOp>(
-      builder.getUnknownLoc(), dataType, builder.getFloatAttr(dataType, 1.0));
-  auto zeroConstantFloatOp = builder.create<ConstantOp>(
-      builder.getUnknownLoc(), dataType, builder.getFloatAttr(dataType, 0.0));
-  block->push_back(oneConstantFloatOp);
-  block->push_back(zeroConstantFloatOp);
+  return;
+}
 
-  // Emit CPU memset function calls.
-  StringRef memsetFuncName;
-  if (dataType == builder.getF32Type()) {
-    memsetFuncName = "mcpuMemset4DFloat";
-  } else if (dataType == builder.getF16Type()) {
-    memsetFuncName = "mcpuMemset4DHalf";
-  } else if (dataType == builder.getBF16Type()) {
-    memsetFuncName = "mcpuMemset4DBF16";
-  }
-  auto mcpuMemset4DFuncOp = FuncOp::create(
-      builder.getUnknownLoc(), memsetFuncName,
+static void launchGPUConvolution(ModuleOp &module, OpBuilder &builder,
+                                 mlir::FloatType dataType, Block *block,
+                                 mlir::AllocOp &filterHostAllocOp,
+                                 mlir::AllocOp &inputHostAllocOp,
+                                 mlir::AllocOp &outputHostAllocOp,
+                                 mlir::MemRefType &filterMemRefType,
+                                 mlir::MemRefType &inputMemRefType,
+                                 mlir::MemRefType &outputMemRefType) {
```

Comment:
Why do you need to pass those additional memref types? Would AllocOp carry those information with them?